### ISSUE TITLE
Introduce other things on the dsl that are missing from WireMockServer

### ISSUE THREAD

**Issue by edeandrea** [wiremock/wiremock-grpc-extension#29]
### Proposal

Many things that I can do with `WireMockServer` are missing here, such as the ability to add a delay.

Now, I could do a PR to just add that feature to the dsl by adding it to the `GrpcStubMappingBuilder` or `GrpcResponseDefinitionBuilder` class, but that would seem to duplicate much of what is in the core `ResponseDefinitionBuilder` class.

There should be a way to expose this. Currently, if you look at https://github.com/wiremock/wiremock-grpc-extension/blob/main/src/main/java/org/wiremock/grpc/dsl/GrpcStubMappingBuilder.java#L64 you can see that `responseBuilder.build` is buried a few layers deep.

This needs to be exposed somewhere so that you can pass it's result into `WireMockGrpcService.stubFor`.

Ideally we should be able to do something like this:

```java
mockGreetingService.stubFor(
        method("greeting")
            .willReturn(
                message(HelloResponse.newBuilder().setGreeting("Hi"))
                    .build()
                    .withFixedDelay(3_000)
                    .with...
        )
);
```

**Comment by edeandrea** [comment]
One solution I played around with was introducing functional callbacks. It would make this nice and easy, but the dsl would have a slight difference in usage from the core WireMock dsl.

Something like this in `WireMockGrpcService`:

```java
  public StubMapping stubFor(GrpcStubMappingBuilder builder) {
    return stubFor(builder, r -> {});
  }

  public StubMapping stubFor(GrpcStubMappingBuilder builder, Consumer<ResponseDefinitionBuilder> builderConsumer) {
    final ResponseDefinitionBuilder responseDefinitionBuilder = builder.buildGrpcMappings();
    builderConsumer.accept(responseDefinitionBuilder);

    final StubMapping stubMapping = builder.build(serviceName, responseDefinitionBuilder);
    wireMock.register(stubMapping);
    return stubMapping;
  }
```

and then it would be used like this:

```java
mockGreetingService.stubFor(
      method("greeting").willReturn(message(HelloResponse.newBuilder().setGreeting("Hi"))),
      builder -> builder.withFixedDelay(10_000)
    );
```

**Comment by edeandrea** [comment]
Delays seem to have been implemented with #31
